tonic/models/vic/grid_params.py

```diff
@@ -917,6 +1150,34 @@ def grid_params(soil_dict, target_grid, snow_dict, veglib_dict, veg_dict,
                 for y, x in pyzip(ymask, xmask):
                     out_dict[var][:, :, y, x] = fill_val
 
+            elif mydict[var].ndim == 4:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                out_dict[var] = np.ma.zeros((j, k, l, ysize, xsize),
+                                            dtype=dtype)
+                for jj in pyrange(j):
+                    for kk in pyrange(k):
+                        for ll in pyrange(l):
+                            out_dict[var][jj, kk, ll, yi, xi] = mydict[var][:, jj, kk, ll]
```

Comment:
I see. This was one of the first things I wrote in Python.  Some of the code in here needs a serious refactor (for reasons like this one).  You don't need to change it but for future reference, we shouldn't need to use a for loop here.